utils/gen-styleguide-prod-config.js

```diff
@@ -180,32 +150,55 @@ function maybeGetPackageInfoForSection(section) {
     }
 }
 
+/**
+ * This function recurses through the sections and sub-sections, removing
+ * any that are marked as private. It returns the modified array with the
+ * private sections removed.
+ */
+function removePrivateSections(sections) {
+    if (!sections) {
+        return sections;
+    }
+
+    for (let i = sections.length; i > 0; i--) {
+        const sectionIndex = i - 1;
+        if (sections[sectionIndex].private) {
+            // eslint-disable-next-line no-console
+            console.log(
+                `Removing private section ${sections[sectionIndex].name}`,
+            );
+
+            // If the section is private, we presume all content of the section is
+            // private too.
+            sections.splice(sectionIndex, 1);
+        } else {
+            // And now, we move into the chilren of the section and repeat.
+            sections[sectionIndex].sections = removePrivateSections(
+                sections[sectionIndex].sections,
+            );
+        }
+    }
+    return sections;
+}
```

Comment:
Moved from before the other functions to after. It ended up not being necessary to do so, but I forgot to move it back.